akka-http-core/src/test/scala/akka/http/impl/engine/client/HttpsProxyGraphStageSpec.scala

```diff
@@ -24,7 +25,35 @@ class HttpsProxyGraphStageSpec extends AkkaSpec {
         sink.request(1)
 
         val connectMsg = flowInProbe.requestNext()
-        connectMsg.utf8String should startWith(s"CONNECT ${targetHostName}:${targetPort} HTTP/1.1")
+        connectMsg.utf8String should startWith(s"CONNECT $targetHostName:$targetPort HTTP/1.1")
+        flowOutProbe.sendNext(ByteString("HTTP/1.0 200 Connection established\r\n\r\n"))
+
+        // we deliberately wait here some time to see if ProxyGraphStage properly handles slow demand from Https Proxy
+        flowInProbe.expectNoMsg(300.millis)
+
+        source.sendNext(ByteString("something"))
+        flowInProbe.requestNext(ByteString("anything"))
+        flowOutProbe.sendNext(ByteString("anything"))
+        sink.expectNext(ByteString("anything"))
+
+        flowInProbe.requestNext(ByteString("something"))
+        flowOutProbe.sendNext(ByteString("something"))
+        sink.requestNext(ByteString("something"))
+
+        source.sendComplete()
+        flowOutProbe.sendComplete()
+        sink.expectComplete()
+      }
+    }
+
+    "send CONNECT message with ProxyAuth header and then forward incoming messages" in new Context {
+      testCaseWithBasicProxyAuth { (source, flowInProbe, flowOutProbe, sink) ⇒
+        source.sendNext(ByteString("anything"))
+        sink.request(1)
+
+        val connectMsg = flowInProbe.requestNext()
+        val proxyHeader = `Proxy-Authorization`(proxyAuth)
+        connectMsg.utf8String should startWith(s"CONNECT $targetHostName:$targetPort HTTP/1.1\r\nHost: $targetHostName\r\n$proxyHeader")
```

Comment:
Let's spell the header out here to make sure it renders correctly (just copying the implementation to the test will make it never fail...). Might make sense to use a multi-line-string with `stripMarginWithNewline`.